File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, int> m;
        int r = 0;
        for (int i : nums){
            if (m[i]) continue;
            r = max(r, m[i] = m[i + m[i+1]] = m[i - m[i-1]] = m[i+1] + m[i-1] + 1);
        }
        return r;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)
        best = 0
        for i in nums:
            if i-1 not in nums:
                j = i + 1
                while j in nums:
                    j += 1
                best = max(best, j - i)
        return best
        